restapi.go

```diff
@@ -2328,3 +2351,103 @@ func (s *Session) RelationshipsMutualGet(userID string) (mf []*User, err error)
 	err = unmarshal(body, &mf)
 	return
 }
+
+// ------------------------------------------------------------------------------------------------
+// Functions specific to application (slash) commands 
+// ------------------------------------------------------------------------------------------------
+
+// ApplicationCommandCreate creates an global application command. If guild is specified - creates guild specific application command.
+func (s *Session) ApplicationCommandCreate(cmd *ApplicationCommand, guildID string) (rcmd *ApplicationCommand, err error) {
+	endpoint := EndpointApplicationGlobalCommands(s.State.User.ID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommands(s.State.User.ID, guildID)
+	}
+	body, err := s.RequestWithBucketID("POST", endpoint, *cmd, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &rcmd)
+	return
+}
+
+// ApplicationCommandDelete deletes global application command by id. If guild is specified - deletes guild specific application command.
+func (s *Session) ApplicationCommandDelete(cmdID, guildID string) (err error) {
+	endpoint := EndpointApplicationGlobalCommand(s.State.User.ID, cmdID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommand(s.State.User.ID, guildID, cmdID)
+	}
+	_, err = s.RequestWithBucketID("DELETE", endpoint, nil, endpoint)
+	return
+}
+
+// ApplicationCommandEdit edits global application command by id and replaces with contents of "cmd". If guild is specified - edits guild specific application command.
+func (s *Session) ApplicationCommandEdit(cmdID, guildID string, cmd *ApplicationCommand) (oldcmd *ApplicationCommand, err error) {
+	endpoint := EndpointApplicationGlobalCommand(s.State.User.ID, cmdID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommand(s.State.User.ID, guildID, cmdID)
+	}
+
+	body, err := s.RequestWithBucketID("PATCH", endpoint, *cmd, endpoint)
```

Comment:
shadow declaration

Reply:
Pl don't user `(err error)`. It's a not good style and u can skip error

Reply:
> shadow declaration

What shadow declaration? it's working perfectly. I didn't shadowed `err` or `body`

Reply:
`err := ` create new local error. Error in this case and all err after will not return. And you skip error check in Request

Reply:
> `err := ` create new local error. Error in this case and all err after will not return. And you skip error check in Request

No, it doesn't check it by yourself on [playground](https://play.golang.org/p/2NLRlQon1QT).

Reply:
As you can tell, this syntax can be confusing so imo we should just declare `body` on the previous line for clarity.

(Go: Variables in the existing scope will not be redeclared by `:=` and will be assigned to instead)